kuka_rox_hardware_interface/src/robot_manager_node.cpp

```diff
@@ -309,21 +319,112 @@ RobotManagerNode::on_deactivate(const rclcpp_lifecycle::State &)
     RCLCPP_ERROR(get_logger(), "Could not stop controllers");
     return ERROR;
   }
+  controller_handler_.ApproveControllerDeactivation();
 
   RCLCPP_INFO(get_logger(), "Successfully stopped controllers");
   return SUCCESS;
 }
 
 bool RobotManagerNode::onControlModeChangeRequest(int control_mode)
 {
-  if (control_mode_map_.find(control_mode) != control_mode_map_.end()) {
+  try {
+    RCLCPP_INFO(get_logger(), "Control mode change process has started");
```

Comment:
Getting to this point doesn't actually mean that the process has started, only the ECS knows that it did. As far as I know here we're waiting for the actual control mode change to occur on the server side, and then change the controllers.

The logs should reflect that as currently it's a bit confusing